### PR TITLE
Lots of improvements to the phpBB3 importer

### PR DESCRIPTION
**Pull request by gschlager**
See the commit messages for more information about the changes.

### CODE REVIEW

**Comment by SamSaffron** [comment]
This is huge :) @nlalonde  when you have a chance can you review? 

@gschlager  we are going to need a bit of time on this.

**Comment by gschlager** [comment]
> This is huge :)

Yeah, sorry about that. :wink:

**Comment by nlalonde** [comment]
Wow. I reviewed the base importer changes, and they look good. Lots of good stuff there, thanks! I didn't review the phpbb importer changes much because I really need to run an import through it to find problems.

**Comment by gschlager** [comment]
I tested it with two small forums and Benno from meta tested it too. So far it seems to work fine. I can't tell how it works with a large forum since I didn't have one to test with.

**Comment by nlalonde** [comment]
Why did you replace so many double-quotes with single quotes with no benefit? It just puts your name on that line in the commit, destroying history.

File: script/import_scripts/base.rb [hunk 1]
```diff
@@ -124,50 +95,48 @@ def reset_site_settings
...
-    @categories_lookup[import_id] || @categories_lookup[import_id.to_s]
+    @lookup.category_id_from_imported_category_id(import_id)
   end
 
   def create_admin(opts={})
     admin = User.new
-    admin.email = opts[:email] || "sam.saffron@gmail.com"
-    admin.username = opts[:username] || "sam"
+    admin.email = opts[:email] || 'sam.saffron@gmail.com'
+    admin.username = opts[:username] || 'sam'
```
Comment (nlalonde) [inline script/import_scripts/base.rb]: Why change "sam" to 'sam'?

**Comment by gschlager** [comment]
Hmm, I guess I applied a Fix in Rubymine's Code Inspector. I can revert those changes if you want.

**Comment by coding-horror** [comment]
Yes I would, since it obfuscates the changes..

**Comment by nlalonde** [comment]
I want to merge this, but those diffs make me cringe...

**Comment by gschlager** [comment]
Okay, I fixed the quotes in `base.rb`.

**Comment by gschlager** [comment]
I fixed a merge conflict. Is there anything else I need to do in order to get this merged?

**Comment by nlalonde** [comment]
Thanks a lot for these improvements!

**Pull request merged**